test/controllers/IncomeTaxComparisonControllerSpec.scala

```diff
@@ -59,55 +61,163 @@ class IncomeTaxComparisonControllerSpec extends PlaySpec
       "not able to fetch comparision details" in {
-        val sut = createSut
+        val controller = new TestController
         when(taxAccountService.taxCodeIncomes(any(), any())(any())).thenReturn(
           Future.successful(TaiNotFoundResponse("Not Found")))
         when(taxAccountService.taxAccountSummary(any(), any())(any())).thenReturn(
           Future.successful(TaiSuccessResponseWithPayload[TaxAccountSummary](taxAccountSummary)))
         when(codingComponentService.taxFreeAmountComponents(any(), any())(any())).thenReturn(
           Future.successful(Seq.empty[CodingComponent]))
 
-        val result = sut.onPageLoad()(RequestBuilder.buildFakeRequestWithAuth("GET"))
+        val result = controller.onPageLoad()(RequestBuilder.buildFakeRequestWithAuth("GET"))
 
         status(result) mustBe INTERNAL_SERVER_ERROR
       }
     }
 
   }
 
+  "rendered CY+1 page" must {
+    "show estimated income for CY and CY+1 for single employment" in {
+      val controller = new TestController
+      when(taxAccountService.taxCodeIncomes(any(), Matchers.eq(TaxYear()))(any())).thenReturn(Future.successful(TaiSuccessResponseWithPayload[Seq[TaxCodeIncome]](taxCodeIncomes)))
+      when(taxAccountService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any())).thenReturn(Future.successful(TaiSuccessResponseWithPayload[Seq[TaxCodeIncome]](taxCodeIncomesCYPlusOne)))
+      when(taxAccountService.taxAccountSummary(any(), any())(any())).thenReturn(Future.successful(TaiSuccessResponseWithPayload[TaxAccountSummary](taxAccountSummary)))
+      when(codingComponentService.taxFreeAmountComponents(any(), any())(any())).thenReturn(Future.successful(Seq.empty[CodingComponent]))
+      when(employmentService.employments(Matchers.any(), Matchers.eq(TaxYear()))(Matchers.any())).thenReturn(Future.successful(Seq(employment)))
+
+      val result = controller.onPageLoad()(RequestBuilder.buildFakeRequestWithAuth("GET"))
+      status(result) mustBe OK
+
+      val doc = Jsoup.parse(contentAsString(result))
+      doc.getElementById("amount-cy-0").text must equal ("£1,111")
+      doc.getElementById("amount-cy-plus-one-0").text must equal ("£2,222")
+    }
+
+    "show estimated income for CY and CY+1 for multiple employments" in {
+      val controller = new TestController
+      when(taxAccountService.taxCodeIncomes(any(), Matchers.eq(TaxYear()))(any())).thenReturn(Future.successful(TaiSuccessResponseWithPayload[Seq[TaxCodeIncome]](taxCodeIncomesMultiple)))
+      when(taxAccountService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any())).thenReturn(Future.successful(TaiSuccessResponseWithPayload[Seq[TaxCodeIncome]](taxCodeIncomesCYPlusOneMultiple)))
+      when(taxAccountService.taxAccountSummary(any(), any())(any())).thenReturn(Future.successful(TaiSuccessResponseWithPayload[TaxAccountSummary](taxAccountSummary)))
+      when(codingComponentService.taxFreeAmountComponents(any(), any())(any())).thenReturn(Future.successful(Seq.empty[CodingComponent]))
```

Comment:
The mocks that return the same things in each of the tests can probably just be done once and this will make the tests shorter and a bit easier to read